spyder/preferences/languageserver.py

```diff
@@ -105,124 +105,128 @@ class LSPServerEditor(QDialog):
-                        'language, along with a executable '
-                        'name for the server to execute '
-                        '(If the instance is local), '
-                        'and the host and port. Finally, '
-                        'you need to provide the '
-                        'arguments that the server accepts. '
-                        'The placeholders <tt>{host}</tt> and '
-                        '<tt>{port}</tt> refer to the host '
-                        'and the port, respectively.')
-        server_settings_description = QLabel(description)
-        server_settings_description.setWordWrap(True)
-
-        lang_label = QLabel(_('Language:'))
+        # Widgets
+        self.server_settings_description = QLabel(description)
+        self.lang_label = QLabel(_('Language:'))
         self.lang_cb = QComboBox(self)
-        self.lang_cb.setToolTip(_('Programming language provided '
-                                  'by the LSP server'))
+        self.external_cb = QCheckBox(_('External server'), self)
+        self.host_label = QLabel(_('Host:'))
+        self.host_input = QLineEdit(self)
+        self.port_label = QLabel(_('Port:'))
+        self.port_spinner = QSpinBox(self)
+        self.cmd_label = QLabel(_('Command to execute:'))
+        self.cmd_input = QLineEdit(self)
+        self.args_label = QLabel(_('Server arguments:'))
+        self.args_input = QLineEdit(self)
+        self.json_label = QLabel(self.JSON_VALID, self)
+        self.conf_label = QLabel(_('LSP Server Configurations:'))
+        self.conf_input = CodeEditor(None)
+
+        self.bbox = QDialogButtonBox(QDialogButtonBox.Ok |
+                                     QDialogButtonBox.Cancel)
+        self.button_ok = self.bbox.button(QDialogButtonBox.Ok)
+        self.button_cancel = self.bbox.button(QDialogButtonBox.Cancel)
+
+        # Widget setup
+        self.setMinimumSize(self.MIN_SIZE)
+        self.setWindowTitle(_('Preferences')
+                            + ': '
+                            + _('Completion and linting')
+                            + ' - '
+                            + _('LSP server editor'))
+
+        self.server_settings_description.setWordWrap(True)
+
+        self.lang_cb.setToolTip(
+            _('Programming language provided by the LSP server'))
         self.lang_cb.addItem(_('Select a language'))
         self.lang_cb.addItems(LSP_LANGUAGES)
 
+        self.button_ok.setEnabled(False)
         if language is not None:
             idx = LSP_LANGUAGES.index(language)
             self.lang_cb.setCurrentIndex(idx + 1)
             self.button_ok.setEnabled(True)
 
-        host_label = QLabel(_('Host:'))
-        self.host_input = QLineEdit(self)
-        self.host_input.setToolTip(_('Name of the host that will provide '
-                                     'access to the server'))
+        self.host_input.setToolTip(
+            _('Name of the host that will provide access to the server'))
```

Comment:
I'm not sure when this is actually visible (all I see is the valid/invalid tooltip, not this one); however, I recommend ``Hostname or IP address of the host on which the server is running``